test/e2e/tests/metrics/attribution/marketing-cookieid.spec.ts

```diff
@@ -0,0 +1,226 @@
+ *
+ * @param mockServer
+ * @returns
+ */
+const mockSegment = async (mockServer: Mockttp) => {
+  return [
+    await mockServer
+      .forPost('https://api.segment.io/v1/batch')
+      .withJsonBodyIncluding({
+        batch: [{ type: 'track', event: 'Main Menu Opened' }],
+      })
+      .thenCallback(() => {
+        return {
+          statusCode: 200,
+        };
+      }),
+  ];
+};
+
+describe('Marketing cookieId', function (this: Suite) {
+  it('update state when the script is executed and the metrics context is updated with the cookieId when an event is fired', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        dappPaths: ['./tests/metrics/attribution/mock-page'],
+        fixtures: new FixtureBuilder()
+          .withMetaMetricsController({
+            metaMetricsId: 'fake-metrics-id',
+            participateInMetaMetrics: true,
+            dataCollectionForMarketing: true,
+          })
+          .build(),
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test?.fullTitle(),
+        testSpecificMock: mockSegment,
+      },
+      async ({
+        driver,
+        mockedEndpoint: mockedEndpoints,
+      }: TestSuiteArguments) => {
+        await unlockWallet(driver);
+
+        await driver.openNewPage(`http://127.0.0.1:8080`);
+        // wait for state to update
+        await driver.delay(5000);
+        await driver.switchToWindowWithTitle(
+          WINDOW_TITLES.ExtensionInFullScreenView,
+        );
+
+        const uiState = await getCleanAppState(driver);
+        assert.equal(uiState.metamask.marketingCampaignCookieId, 12345);
+
+        await driver.clickElement(selectors.accountOptionsMenuButton);
+        const events = await getEventPayloads(
+          driver,
+          mockedEndpoints as MockedEndpoint[],
+        );
+        assert.equal(events.length, 1);
+        const eventContext = events[0].context;
+        assert.equal(eventContext.marketingCampaignCookieId, 12345);
+      },
+    );
+  });
+  it('doesnot update state when dataCollectionForMarketing is already false ', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        dappPaths: ['./tests/metrics/attribution/mock-page'],
+        fixtures: new FixtureBuilder()
+          .withMetaMetricsController({
+            metaMetricsId: 'fake-metrics-id',
+            participateInMetaMetrics: true,
+          })
+          .build(),
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test?.fullTitle(),
+        testSpecificMock: mockSegment,
+      },
+      async ({
+        driver,
+        mockedEndpoint: mockedEndpoints,
+      }: TestSuiteArguments) => {
+        await unlockWallet(driver);
+
+        await driver.openNewPage(`http://127.0.0.1:8080`);
+        // wait for state to update
+        await driver.delay(5000);
+        await driver.switchToWindowWithTitle(
+          WINDOW_TITLES.ExtensionInFullScreenView,
+        );
```

Comment:
I would suggest moving this to the step before opening dapp so we started tracking window.postMessage